src/openrct2/paint/tile_element/Paint.Surface.cpp

```diff
@@ -1125,9 +1099,17 @@ void PaintSurface(PaintSession& session, uint8_t direction, uint16_t height, con
         {
             imageId = ImageId(SPR_TERRAIN_TRACK_DESIGNER);
         }
-        else
+        else if (surfaceObject != nullptr)
         {
-            imageId = GetSurfaceImage(session, surfaceObject, image_offset, rotation, grassLength, showGridlines, false);
+            uint8_t grassLength = 0xFF;
+            if (zoomLevel <= ZoomLevel{ 0 })
```

Comment:
This check is in the original RCT. But I'm wondering if its worth keeping? It just stops the grass appearing when you zoom out. It makes no difference to the code run in this section because the full list of specials is iterated either way. The only thing it might do is marginally improve cache performance by drawing less unique sprites.

I removed the check and the game looked fine, arguably better since now the long grass doesn't suddenly disappear when you zoom out so it might be worth removing this. Thoughts?


Note I have moved this block here since the grassLength is only used for the GetImageId call.

Reply:
I think we can probably ditch it, its not going to make any major difference